core/frontend/render-modes/render-mode-base.php

```diff
@@ -0,0 +1,94 @@
+<?php
+namespace Elementor\Core\Frontend\RenderModes;
+
+use Elementor\Plugin;
+use Elementor\Core\Base\Document;
+use Elementor\Core\Frontend\Render_Mode_Manager;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+abstract class Render_Mode_Base implements Render_Mode_Interface {
+	const ENQUEUE_SCRIPTS_PRIORITY = 10;
+	const ENQUEUE_STYLES_PRIORITY = 10;
+
+	/**
+	 * @var int
+	 */
+	protected $post_id;
+
+	/**
+	 * @var Document
+	 */
+	protected $document;
+
+	/**
+	 * Render_Mode_Base constructor.
+	 *
+	 * @param $post_id
+	 */
+	public function __construct( $post_id ) {
+		$this->post_id = intval( $post_id );
+	}
+
+	/**
+	 * @param $post_id
+	 *
+	 * @return string
+	 */
+	public static function get_url( $post_id ) {
+		return Render_Mode_Manager::get_base_url( $post_id, static::get_name() );
+	}
+
+	/**
+	 * By default do not do nothing.
+	 */
+	public function prepare_render() {
+		add_action( 'wp_enqueue_scripts', [ $this, 'enqueue_scripts' ], static::ENQUEUE_SCRIPTS_PRIORITY );
+		add_action( 'wp_enqueue_scripts', [ $this, 'enqueue_styles' ], static::ENQUEUE_STYLES_PRIORITY );
+	}
+
+	/**
+	 * By default do not do nothing.
+	 */
+	public function enqueue_scripts() {
+		//
+	}
+
+	/**
+	 * By default do not do nothing.
+	 */
+	public function enqueue_styles() {
+		//
+	}
+
```

Comment:
Why do you need to implement an empty method that has already been implemented in the interface?

Reply:
In the interface it is only a declaration and not an implementation.

Reply:
The idea of an interface is forcing. If you don't want to force, you shouldn't put it inside the interface.